chia/wallet/wallet_state_manager.py

```diff
@@ -1474,32 +1474,30 @@ async def coin_added(
             wallet_id,
             wallet_type,
         )
-        farmer_reward = False
-        pool_reward = False
-        if self.is_farmer_reward(height, coin):
-            farmer_reward = True
-        elif self.is_pool_reward(height, coin):
-            pool_reward = True
-
-        farm_reward = False
-        parent_coin_record: Optional[WalletCoinRecord] = await self.coin_store.get_coin_record(coin.parent_coin_info)
-        if parent_coin_record is not None and wallet_type.value == parent_coin_record.wallet_type:
-            change = True
+
+        if self.is_pool_reward(height, coin):
+            tx_type = TransactionType.COINBASE_REWARD
+        elif self.is_farmer_reward(height, coin):
```

Comment:
I suppose a single coin can't be both `pool_reward` and `farmer_reward`, right? if so, the order of checking here would matter

Reply:
prior to this patch, if `is_farmer_reward()` is true we don't ask about `is_pool_reward()`, so `tx_type` becomes `TransactionType.FEE_REWARD.value`.

In your patch, if `is_pool_reward()` is true, you don't ask `is_farmer_reward()`, so `tx_type` becomes `TransactionType.COINBASE_REWARD`.

If a coin could be both, this would be a change in behavior

Reply:
I mean, how can a coin be both?

Reply:
I'm not arguing it can. I'm arguing that this transformation is not mechanical, but requires this prerequisite knowledge